decisions/deprecated-api.md

```diff
@@ -0,0 +1,40 @@
+# Deprecated API
```

Comment:
is `decisions` the right folder for this?

Reply:
I was also a little hesitant, there is not another good place to put documents like this AFAIK and I didn't feel like creating another dir for this. @anambl leaned toward placing these documents here in the ticket.